### PR TITLE
[ci][terraform][aws] tags with metadata

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Standardise the below tags/labels in the terraform resources for AWS/GCP:

- `environment` => static value
- `repo`.              => static value
- `branch`           => dynamic value
- `build`              => dynamic value 
- `created_date` => dynamic value

## Why is it important?

Help with tearing down any of the ephemeral resources which were not successfully removed as part of the system tests.

## Implementation details

* Those tag/labels are lowercase based to be [GCP](https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements)/[AWS](https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html#tag-conventions) compliance
* `environment=CI` will allow to filter all those cloud resources which were created in the CI
* While `created_date` will help to filter those which were older than X days or X hours, since it's Unix epoch time based.
* `branch` and `build` will help to identify what build in the CI and what PR, Branch or Tag created those resources.

## Further details

> BUILD_ID
The current build ID, identical to BUILD_NUMBER for builds created in 1.597+, but a YYYY-MM-DD_hh-mm-ss timestamp for older builds.

> BRANCH_NAME
For a multibranch project, this will be set to the name of the branch being built, for example in case you wish to deploy to production from master but not from feature branches; if corresponding to some kind of change request, the name is generally arbitrary (refer to CHANGE_ID and CHANGE_TARGET).

## Issues

Requires https://github.com/elastic/elastic-package/pull/792

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -258,7 +258,11 @@ def withCloudTestEnv(Closure body) {
     [var: "AWS_SECRET_ACCESS_KEY", password: aws.secret_key],
   ])
   withEnvMask(vars: maskedVars) {
-    body()
+    withEnv(["TF_VAR_BUILD_ID=${BUILD_ID}",
+             "TF_VAR_BRANCH_NAME=${BRANCH_NAME}",
+             "TF_VAR_CREATED_DATE=${current.getRawBuild().getTimestampString2()}"]) {
```
Comment (v1v) [inline .ci/Jenkinsfile:263]:
```suggestion
             "TF_VAR_CREATED_DATE=${currentBuild.getRawBuild().getTimestampString2()}"]) {
```

**Review by mtojek — CHANGES_REQUESTED** [review]
I'm setting a blocker on this to merge the https://github.com/elastic/elastic-package/pull/792 first, then iterate on this one.

**Review by mtojek — COMMENTED** [review]
As the [PR](https://github.com/elastic/elastic-package/pull/792) changed the implementation of elastic-package, let's first [release a new version](https://github.com/elastic/elastic-package#release-process). It's just about Git tagging the latest commit in the main branch.

**Review by kuisathaverat — APPROVED** [review]

**Comment by v1v** [comment]
I just merged the PR for the `elastic-package` and created the tag https://github.com/elastic/elastic-package/releases/tag/v0.47.0

**Comment by mtojek** [comment]
Version bump PR here: https://github.com/elastic/integrations/pull/3182

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -258,7 +259,13 @@ def withCloudTestEnv(Closure body) {
     [var: "AWS_SECRET_ACCESS_KEY", password: aws.secret_key],
   ])
   withEnvMask(vars: maskedVars) {
-    body()
+    withEnv(["TF_VAR_BRANCH=${env.BRANCH_NAME.toLowerCase()}",
```
Comment (v1v) [inline .ci/Jenkinsfile:262]: For some reason these changes are not consumed:

```
terraform_1  |       + tags                                 = {
terraform_1  |           + "Name" = "elastic-package-test-33598"
terraform_1  |         }
terraform_1  |       + tags_all                             = {
terraform_1  |           + "Name"         = "elastic-package-test-33598"
terraform_1  |           + "branch"       = "unknown-branch"
terraform_1  |           + "build"        = "unknown-build"
terraform_1  |           + "created_date" = "unknown-date"
terraform_1  |           + "environment"  = "unknown-environment"
terraform_1  |           + "repo"         = "unknown-repo-name"
terraform_1  |         }
```

**Comment by mtojek** [comment]
@v1v Please merge the main branch, so that this PR will pick the last bumped version.

**Comment by v1v** [comment]
@mergifyio rebase

File: .ci/Jenkinsfile (hunk 2 above)
Comment (v1v) [reply .ci/Jenkinsfile:262]: Somehow some env variables are not populated from the CI:

```
terraform_1  |       + tags                                 = {
terraform_1  |           + "Name" = "elastic-package-test-17719"
terraform_1  |         }
terraform_1  |       + tags_all                             = {
terraform_1  |           + "Name"         = "elastic-package-test-17719"
terraform_1  |           + "branch"       = "unknown"
terraform_1  |           + "build"        = "6"
terraform_1  |           + "created_date" = "unknown"
terraform_1  |           + "environment"  = "unknown"
terraform_1  |           + "repo"         = "integrations"
```

I guess, the elastic-packages has a higher precendence:
- https://github.com/elastic/elastic-package/pull/792/files#diff-1d4bd2c1ad56472b4dc4c97566e6ce9dc00bda82eb59cf14d1ab8d5c930a1fe0R9-R14

Comment (mtojek) [reply .ci/Jenkinsfile]: I would start with investigating why the releasing job for v0.47.0 failed: https://beats-ci.elastic.co/job/Ingest-manager/job/elastic-package/job/v0.47.0/

Comment (mtojek) [reply .ci/Jenkinsfile]: @v1v It looks like we need a fix:

```
terraform_1  | Plan: 1 to add, 0 to change, 0 to destroy.
terraform_1  | google_compute_instance.default: Creating...
terraform_1  |
terraform_1  | Error: Error creating instance: googleapi: Error 400: Invalid value for field 'resource.labels': ''. Label value 'v0.47.0' violates format constraints. The value can only contain lowercase letters, numeric characters, underscores and dashes. The value can be at most 63 characters long. International characters are allowed., invalid
terraform_1  |
terraform_1  |   with google_compute_instance.default,
terraform_1  |   on compute.tf line 7, in resource "google_compute_instance" "default":
terraform_1  |    7: resource "google_compute_instance" "default" {
terraform_1  |
terraform_1  | + echo 'Terraform definitions applied.'
terraform_1  | Terraform definitions applied.
terraform_1  | + set +x
```

Comment (v1v) [reply .ci/Jenkinsfile]: Fixed already added to the elastic-package and also here, so the new metadata is populated now

```
terraform_1  |       + tags                                 = {
terraform_1  |           + "Name" = "elastic-package-test-73355"
terraform_1  |         }
terraform_1  |       + tags_all                             = {
terraform_1  |           + "Name"         = "elastic-package-test-73355"
terraform_1  |           + "branch"       = "pr-3126"
terraform_1  |           + "build"        = "8"
terraform_1  |           + "created_date" = "1650891241341"
terraform_1  |           + "environment"  = "ci"
terraform_1  |           + "repo"         = "integrations"
terraform_1  |         }
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -5,7 +5,11 @@
 pipeline {
   agent { label 'ubuntu-20 && immutable' }
   environment {
-    BASE_DIR="src/github.com/elastic/integrations"
+    BRANCH_NAME_LOWER_CASE = "${env.BRANCH_NAME.toLowerCase().replaceAll('\\.', '_')}"
```
Comment (mtojek) [inline .ci/Jenkinsfile:8]: I guess that we have to apply a consistent fix as we agree on https://github.com/elastic/elastic-package/pull/799

Comment (v1v) [reply .ci/Jenkinsfile:8]:
```suggestion
    BRANCH_NAME_LOWER_CASE = "${env.BRANCH_NAME.toLowerCase().replaceAll('[^a-z0-9-]', '-')}"
```

**Comment by v1v** [comment]
> Just double-checking, system tests for GCP are not there yet, right?

I could not find any other references to terraform but only in the `aws` package

**Comment by mtojek** [comment]
I checked these errors and they're unrelated. I guess that you can merge this PR.

**Pull request merged**